src/eval-frame/components/reps/rep-utils/rep-type-chooser.js

```diff
@@ -0,0 +1,35 @@
+import _ from "lodash";
+
+const ROWS_TO_CHECK = 100;
+
+function sameKeys(a, b) {
+  if (a === b) return true;
+  if (a == null || b == null) return false;
+  const aKeys = Object.keys(a);
+  const bKeys = Object.keys(b);
+  if (aKeys.length !== bKeys.length) return false;
+
+  aKeys.sort();
+  bKeys.sort();
+
+  for (let i = 0; i < aKeys.length; ++i) {
+    if (aKeys[i] !== bKeys[i]) return false;
+  }
+  return true;
+}
+
+export function isRowDf(obj) {
+  if (!_.isArray(obj) || obj.length < 2 || !_.isPlainObject(obj[0])) {
+    return false;
+  }
+  const rowsToCheck = Math.min(ROWS_TO_CHECK, _.size(obj));
+  for (let i = 1; i < rowsToCheck; i++) {
+    if (!_.isPlainObject(obj[i])) {
```

Comment:
Could this be made to also check for an array of small arrays?  That would cover the case where the data is from a CSV without a header row, or a Numpy structured array.

Reply:
i'd prefer to keep the semantics very clear about what these objects are, b/c i think it will make the code clearer and the testing easier, and will also allow us to keep fine-grained control on what components we're using and make sure we're using them correctly. for example a `rowDf` is always an array of >1 plain objects that have all have identical keys, and it is meant to be shown in a tabular view with column headings, so passing in an array of arrays would break the expectations of the component downstream.

i think it will be easy to expand out from here in future PRs to add things like columnar arrays, arrow objects, tablular things without column names, etc. but we'll want to match those in separate type-chooser functions to make sure that we're handling them correctly in the corresponding component